metrics/framework/Metrics.cpp

```diff
@@ -0,0 +1,212 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "Metrics.hpp"
+
+using namespace hpccMetrics;
+
+void CounterMetric::getReportValues(MeasurementVector &mv) const
+{
+    mv.emplace_back((std::make_shared<Measurement<uint32_t>>(name, description, ValueType::METRICS_INTEGER, count.load())));
+}
+
+
+void GaugeMetric::getReportValues(MeasurementVector &mv) const
+{
+    mv.emplace_back((std::make_shared<Measurement<float>>(name, description, ValueType::METRICS_FLOAT, gaugeValue.load())));
+}
+
+
+IMetricSink *MetricSink::getSinkFromLib(const char *type, const char *sinkName, const IPropertyTree *pSettingsTree)
```

Comment:
I'm not sure why this would be a member of MetricSink, rather than a member of the class that calls it.

Reply:
I'm not sure why either. It was copied from an older implementation since the mechanics of what it did has not changed. I moved the code to a single static method in the reporter class.